server/app/forms/QuestionForm.java

```diff
@@ -53,49 +59,64 @@ protected QuestionForm(QuestionDefinition qd) {
     }
   }
 
-  public String getQuestionName() {
+  public final String getQuestionName() {
     return questionName;
   }
 
-  public void setQuestionName(String questionName) {
+  public final void setQuestionName(String questionName) {
     this.questionName = checkNotNull(questionName);
   }
 
-  public String getQuestionDescription() {
+  public final String getQuestionDescription() {
     return questionDescription;
   }
 
-  public void setQuestionDescription(String questionDescription) {
+  public final void setQuestionDescription(String questionDescription) {
     this.questionDescription = checkNotNull(questionDescription);
   }
 
-  public Optional<Long> getEnumeratorId() {
+  public final Optional<Long> getEnumeratorId() {
     return enumeratorId;
   }
 
-  public void setEnumeratorId(String enumeratorId) {
+  public final void setEnumeratorId(String enumeratorId) {
     this.enumeratorId =
         enumeratorId.isEmpty() ? Optional.empty() : Optional.of(Long.valueOf(enumeratorId));
   }
 
   public abstract QuestionType getQuestionType();
 
-  public String getQuestionText() {
+  public final String getQuestionText() {
     return questionText;
   }
 
-  public void setQuestionText(String questionText) {
+  public final void setQuestionText(String questionText) {
     this.questionText = checkNotNull(questionText);
   }
 
-  public String getQuestionHelpText() {
+  public final String getQuestionHelpText() {
     return questionHelpText;
   }
 
-  public void setQuestionHelpText(String questionHelpText) {
+  public final void setQuestionHelpText(String questionHelpText) {
     this.questionHelpText = checkNotNull(questionHelpText);
   }
 
+  public final String getRedirectUrl() {
+    if (redirectUrl == null) {
+      return "";
+    }
+    if (!redirectUrl.isBlank() && URI.create(redirectUrl).isAbsolute()) {
```

Comment:
Interesting, you decided to do validation in getter instead of setter? Is that to be safer in case someone else sets value outside of the main `setRedirecUrl`? It will make debugging harder though.

Reply:
Added a comment. Correct. The goal is to prevent any bad values from finding their way out of the class and the getter seemed the safest place to do this. My concern with doing this in the setter is that there are multiple ways to get these values set that may change over time (e.g. a new constructor that may initialize the value directly, Play claiming it supports building forms with reflection, etc). For this one, it seemed better to play it safe by preventing the value from getting out. In all of these cases, the real way we'd see any errors is through someone directly manipulating HTTP requests.